ktor-server/ktor-server-plugins/ktor-server-call-logging/jvm/src/io/ktor/server/plugins/callloging/CallLogging.kt

```diff
@@ -5,265 +5,92 @@
-         * processing.
-         */
-        public fun mdc(name: String, provider: (ApplicationCall) -> String?) {
-            mdcEntries.add(MDCEntry(name, provider))
-        }
-
-        /**
-         * Configure application call log message.
-         */
-        public fun format(formatter: (ApplicationCall) -> String) {
-            formatCall = formatter
-        }
-
-        /**
-         * Disables colors in log message in case the default formatter was used.
-         * */
-        public fun disableDefaultColors() {
-            isColorsEnabled = false
-        }
-
-        private fun defaultFormat(call: ApplicationCall): String =
-            when (val status = call.response.status() ?: "Unhandled") {
-                HttpStatusCode.Found -> "${colored(status as HttpStatusCode)}: " +
-                    "${call.request.toLogStringWithColors()} -> ${call.response.headers[HttpHeaders.Location]}"
-                "Unhandled" -> "${colored(status, Ansi.Color.RED)}: ${call.request.toLogStringWithColors()}"
-                else -> "${colored(status as HttpStatusCode)}: ${call.request.toLogStringWithColors()}"
-            }
-
-        internal fun ApplicationRequest.toLogStringWithColors(): String =
-            "${colored(httpMethod.value, Ansi.Color.CYAN)} - ${path()}"
-
-        private fun colored(status: HttpStatusCode): String {
-            try {
-                if (!AnsiConsole.isInstalled()) {
-                    AnsiConsole.systemInstall()
-                }
-            } catch (cause: Throwable) {
-                isColorsEnabled = false // ignore colors if console was not installed
-            }
-
-            return when (status) {
-                HttpStatusCode.Found, HttpStatusCode.OK, HttpStatusCode.Accepted, HttpStatusCode.Created -> colored(
-                    status,
-                    Ansi.Color.GREEN
-                )
-                HttpStatusCode.Continue, HttpStatusCode.Processing, HttpStatusCode.PartialContent,
-                HttpStatusCode.NotModified, HttpStatusCode.UseProxy, HttpStatusCode.UpgradeRequired,
-                HttpStatusCode.NoContent -> colored(
-                    status,
-                    Ansi.Color.YELLOW
-                )
-                else -> colored(status, Ansi.Color.RED)
-            }
-        }
-
-        private fun colored(value: Any, color: Ansi.Color): String =
-            if (isColorsEnabled) Ansi.ansi().fg(color).a(value).reset().toString()
-            else value.toString() // ignore color
+public val CallLogging: ApplicationPlugin<Application, CallLoggingConfig, PluginInstance> = createApplicationPlugin(
+    "CallLogging",
+    ::CallLoggingConfig
+) {
+    val log = pluginConfig.logger ?: application.log
+    val filters = pluginConfig.filters
+    val formatCall = pluginConfig.formatCall
+
+    fun log(message: String) = when (pluginConfig.level) {
```

Comment:
Nit: maybe just private methods would be better here?

Reply:
they are inside lambda and internal to the plugin. Can't make outside because of capturing

Reply:
You can pass parameters cuz there's only a few of then required